torch/_dynamo/source.py

```diff
@@ -901,6 +902,7 @@ def is_from_source(source: Source, target: Source):
     return source == target
 
 
+@functools.lru_cache(None)
```

Comment:
Unlimited LRU caches are a bit much for these functions, isn't it?

Reply:
Especially since you are caching on the "instance" and not the type

Reply:
Thanks for catching this. Did not pay attention to the autocomplete.